### PR TITLE
Add io.github.aldrinzigmundv.InnerMirror

### PR DESCRIPTION
**Pull request by aldrinzigmundv**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.github.aldrinzigmundv.InnerMirror.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "app-id": "com.github.aldrinzigmundv.InnerMirror",
```
Comment (hfiguiere) [inline com.github.aldrinzigmundv.InnerMirror.json:2]: it must be `io.github` not `com.github`

**Comment by hfiguiere** [comment]
bot, build io.github.aldrinzigmundv.InnerMirror

**Comment by hfiguiere** [comment]
cargo must be run in --offline mode, and apparently isn't

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "command": "innermirror",
+    "finish-args" : [
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=ipc"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin"
+    },
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:16]: you must set CARGO_HOME to `/run/build/innermirror/cargo`

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "modules": [
+        {
+            "name": "innermirror",
+            "buildsystem": "meson",
+            "run-tests": true,
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:21]: this might actually not work.

File: io.github.aldrinzigmundv.InnerMirror.json (hunk 2 above)
Comment (hfiguiere) [reply io.github.aldrinzigmundv.InnerMirror.json:19]: this is not what you set it to. it actually matters.

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+                "cargo --offline build --release --verbose",
+                "install -Dm755 ./target/debug/rust-flatpak -t /app/bin/",
+                "install -Dm644 ./data/${FLATPAK_ID}.metainfo.xml.in.in -t /app/share/metainfo/",
+                "install -Dm644 ./data/${FLATPAK_ID}.desktop.in.in -t /app/share/applications/",
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:30]: this is invalid for a desktop file. the `.in.in` is supposed to bo processeded twice by the meson build script.

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        "env" : {
+            "CARGO_HOME" : "/run/build/rust-flatpak/cargo"
+        }
+    },
+    "modules": [
+        {
+            "name": "innermirror",
+            "buildsystem": "meson",
+            "run-tests": true,
+            "build-commands": [
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:25]: buildsystem is meson so build-commands is ignored.

Comment (hfiguiere) [reply io.github.aldrinzigmundv.InnerMirror.json:25]: It actually should just work to use meson. You should even be able vendor the crates in the tarball and not need cargo-sources....

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    "finish-args" : [
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=ipc"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "env" : {
+            "CARGO_HOME" : "/run/build/rust-flatpak/cargo"
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:17]:
```suggestion
            "CARGO_HOME" : "/run/build/innermirror/cargo"
```

**Comment by hfiguiere** [comment]
bot, build bot, build io.github.aldrinzigmundv.InnerMirror

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        }
+    },
+    "modules": [
+        {
+            "name": "innermirror",
+            "buildsystem": "simple",
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+                "cargo --offline build --release --verbose",
+                "install -Dm755 ./target/debug/rust-flatpak -t /app/bin/",
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:27]: you build `--release` so there is nothing there.

File: io.github.aldrinzigmundv.InnerMirror.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "env" : {
+            "CARGO_HOME" : "/run/build/innermirror/cargo"
+        }
+    },
+    "modules": [
+        {
+            "name": "innermirror",
+            "buildsystem": "simple",
```
Comment (hfiguiere) [inline io.github.aldrinzigmundv.InnerMirror.json:23]: why do you not use the meson build system?

**Comment by aldrinzigmundv** [comment]
Thank you for your guidance and support. After some careful consideration, I've decided to take a step back from the app submission process given my ongoing commitments to other projects. I believe that a more thorough review is necessary to ensure the best possible outcome.

**Pull request closed without merging**